Mathlib/Topology/Algebra/Affine.lean

```diff
@@ -4,70 +4,85 @@ Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Frédéric Dupuis
 -/
 import Mathlib.LinearAlgebra.AffineSpace.AffineMap
-import Mathlib.Topology.Algebra.MulAction
-import Mathlib.Topology.Algebra.Group.Defs
+import Mathlib.Topology.Algebra.Group.AddTorsor
 
 /-!
 # Topological properties of affine spaces and maps
 
-For now, this contains only a few facts regarding the continuity of affine maps in the special
-case when the point space and vector space are the same.
-
-TODO: Deal with the case where the point spaces are different from the vector spaces. Note that
-we do have some results in this direction under the assumption that the topologies are induced by
-(semi)norms.
+This file contains a few facts regarding the continuity of affine maps.
 -/
 
 
 namespace AffineMap
 
-variable {R E F : Type*}
-variable [AddCommGroup E] [TopologicalSpace E]
-variable [AddCommGroup F] [TopologicalSpace F] [IsTopologicalAddGroup F]
+variable
+  {R V P W Q : Type*}
+  [AddCommGroup V] [TopologicalSpace V]
+  [AddTorsor V P] [TopologicalSpace P] [IsTopologicalAddTorsor P]
+  [AddCommGroup W] [TopologicalSpace W]
+  [AddTorsor W Q] [TopologicalSpace Q] [IsTopologicalAddTorsor Q]
 
 section Ring
 
-variable [Ring R] [Module R E] [Module R F]
+variable [Ring R] [Module R V] [Module R W]
+
+/-- If `f` is an affine map, then its linear part is continuous iff `f` is continuous. -/
+theorem continuous_linear_iff {f : P →ᵃ[R] Q} : Continuous f.linear ↔ Continuous f := by
+  inhabit P
+  have :
+    (f.linear : V → W) =
+      (Homeomorph.vaddConst <| f default).symm ∘ f ∘ (Homeomorph.vaddConst default) := by
+    ext v
+    simp
+  rw [this]
+  simp only [Homeomorph.comp_continuous_iff, Homeomorph.comp_continuous_iff']
 
 /-- An affine map is continuous iff its underlying linear map is continuous. See also
 `AffineMap.continuous_linear_iff`. -/
-theorem continuous_iff {f : E →ᵃ[R] F} : Continuous f ↔ Continuous f.linear := by
-  constructor
-  · intro hc
-    rw [decomp' f]
-    exact hc.sub continuous_const
-  · intro hc
-    rw [decomp f]
-    exact hc.add continuous_const
```

Comment:
Note that there is a slight loss in generality here because the old lemma assume no compatibility between the topology and algebra on `E`. Of course you can't be as general in your setup because you need some link between the topology of `V` and `P` (this role is assumed by the identity being continuous when `V = P`).

I don't think it really matters, but as a middle ground, could you assume only `ContinuousConstVAdd V P`? In fact, I think you can also generalize to `ContinuousConstVAdd W Q` and it should still work?

Reply:
In fact I think these assumptions apply to the whole file. I haven't thought hard about it though.

Reply:
It randomly struck me that I was speaking nonsense above: `ContinuousConstVAdd` is the continuity `P -> P` not `V -> P` :facepalm: 

You can forget these comments...